docarray/array/document.py

```diff
@@ -146,11 +146,13 @@ def __getitem__(
                 and len(index) == 2
                 and isinstance(index[0], (slice, Sequence))
             ):
+                # edge case where 2 ids are passed
+                if isinstance(index[0], str) and index[0] in self._id2offset:
```

Comment:
this logic itself is also problematic, it does not work with column selector https://docarray.jina.ai/fundamentals/documentarray/access-attributes/

Reply:
you can try `da['2', 'id']` under your branch it will break

Reply:
I would suggest a completely readthrough on "access attributes" chapter before fixing edge cases like this

Reply:
Now it should be fixed. There is still the possibility that an id has exactly the same str than an element_selector or a column selector in which case I guess the notation could be ambiguous as Nan mentions below. 

I think this is a 'very stretched edge case'  but it's a possibility:

```
from docarray import Document, DocumentArray

da = DocumentArray([Document(id='@r', text='a1'),
                    Document(id='2', text='a2'),
                    Document(id='3', text='a3')])
```

Then `da['@r']` should be a single item or a `DocumentArray` with all root items?

We could assume `id` strings do not start with `@`, otherwise the logic in getitem/setitem  is ambiguous. To me this is a reasonable assumption but I guess we should make a `Note` in the docs?